### ISSUE TITLE
v0.11.2 doctests fail with Rust 1.78+ due to violated unsafe preconditions on x86_64-unknown-linux-gnu

### ISSUE THREAD

**Issue by decathorpe** [nabijaczleweli/safe-transmute-rs#69]
Since the update to Rust 1.78 (which enabled unsafe precondition checks in std under more circumstances), there are a number of doctests that now fail due to un-aligned or non-null pointers:

    src/base.rs - base::from_bytes (line 38)
    src/base.rs - base::from_bytes_pedantic (line 77)
    src/base.rs - base::transmute_many (line 121)
    src/base.rs - base::transmute_many_permissive (line 208)
    src/trivial.rs - trivial::transmute_trivial (line 182)
    src/trivial.rs - trivial::transmute_trivial_many (line 255)
    src/trivial.rs - trivial::transmute_trivial_many_mut (line 292)
    src/trivial.rs - trivial::transmute_trivial_pedantic (line 220)

The error messages are exactly the same for all of them (stack trace is long and not very interesting):

```
Test executable failed (signal: 6 (SIGABRT) (core dumped)).
stderr:
thread 'main' panicked at library/core/src/panicking.rs:156:5:
unsafe precondition(s) violated: slice::from_raw_parts requires the pointer to be aligned and non-null, and the total size of the slice not to exceed `isize::MAX`
```

**Comment by decathorpe** [comment]
I got this failure on x86_64-unknown-linux-gnu (see issue title) - with Rust 1.78.0 (stable), on Fedora.

I forgot to mention that the failure is happening in v0.11.2, but not in git master. So this commit looks like it fixes the failure: https://github.com/nabijaczleweli/safe-transmute-rs/commit/6e9520a6d9acdf5eac96b5dc8a15aa241cd2e0b7

Would it be possible to get a new release with those fixes? Or should I backport them to the Fedora package?

**Comment by nabijaczleweli** [comment]
Released as [`v0.11.3`](https://github.com/nabijaczleweli/safe-transmute-rs/releases/tag/v0.11.3)

**Comment by decathorpe** [comment]
Awesome, thanks!
